### PR TITLE
feat!: move readside configuration to yaml file

### PR DESCRIPTION
**Pull request by Tochemey**
The feature has been successfully QA in this PR(https://github.com/chief-of-state/cos-python-sample/pull/2)
@zenyui let us get this in.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -67,26 +67,46 @@ The following options can be configured via environment variables.
...
+port: 50054
+useTls: false
+autoStart: true
+```
 
-| environment variable                                | description                     | default |
-|-----------------------------------------------------|---------------------------------|---------|
-| COS_READ_SIDE_CONFIG__<SETTING_NAME>__<READSIDE_ID> | readside configuration settings | <none>  |
+##### Note
+As you can see in the sample read side config one can specify multiple read side in the yaml file using the `---` yaml
```
Comment (zenyui) [inline docs/configuration.md:103]: Interesting. Why not use a YAML array instead?

**Review by zenyui — COMMENTED** [review]
Overall very cool. 2 thoughts:
- let's discuss if we really need to backward break the config 
- I think we should make each file a yaml array instead of using the `---` separators, thoughts?

File: docs/configuration.md (hunk 1 above)
Comment (Tochemey) [reply docs/configuration.md:103]: Making use of the YAML array I think will be great. However I don't think we should be backward compatible. This feature is easy to adapt AFAIK. This is going to be a breaking change in how to setup the read sides

Comment (Tochemey) [reply docs/configuration.md:103]: My goal is to ge rid of the env var which is error prone

**Comment by Tochemey** [comment]
> Overall very cool. 2 thoughts:
> 
> * let's discuss if we really need to backward break the config
> * I think we should make each file a yaml array instead of using the `---` separators, thoughts?

@zenyui Are you suggesting we have for each readside we should have a yaml file instead of a single yaml file containing all read side configs?

File: docs/configuration.md (hunk 1 above)
Comment (Tochemey) [reply docs/configuration.md:103]: > Interesting. Why not use a YAML array instead?

The `---` is a yaml file document separator. So my goal is to have a single file containing all read side configs

**Comment by zenyui** [comment]
> > Overall very cool. 2 thoughts:
> 
> > 
> 
> > * let's discuss if we really need to backward break the config
> 
> > * I think we should make each file a yaml array instead of using the `---` separators, thoughts?
> 
> 
> 
> @zenyui Are you suggesting we have for each readside we should have a yaml file instead of a single yaml file containing all read side configs?

@Tochemey i was suggesting nesting them in a yaml array instead of separate yaml statements with the `---` document separator. With that said, I think your solution is cool.

**Comment by Tochemey** [comment]
@zenyui I have refactored the code to read either a folder containing the read side configuration files or a single yaml file containing all the read side configurations.

**Pull request merged**